src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -102,6 +102,8 @@ def configure_common_settings(cli_ctx, client):
     command_name_suffix = ';completer-request' if cli_ctx.data['completer_active'] else ''
     client._client.add_header('CommandName',  # pylint: disable=protected-access
                               "{}{}".format(cli_ctx.data['command'], command_name_suffix))
+    client._client.add_header('ParameterSetName',  # pylint: disable=protected-access
+                              "{}".format(cli_ctx.data['safe_params']))
```

Comment:
this can just be `cli_ctx.data['safe_params']`

Reply:
This fails trying to convert a list to string ("expected string or bytes-like object"), using format seemed like an easy way to 'cast' it.

Reply:
Saw the ''.join method being more common, so switched to that.

Reply:
Apologize for the wrong comment at the beginning as i don't know the data types, but please do `' '.join(cli_ctx.data['safe_params'])` as that would avoid the "[]" brought in by the str.
Also, please rebase from the latest dev which i hope would get rid of the seemingly irrelevant CI error

Reply:
Ci passes now. This means something went wrong with using ''.join(). The headers were perhaps malformed.

Reply:
It is a `None ref` error